docs/libcurl/opts/CURLOPT_ECH.md

```diff
@@ -0,0 +1,72 @@
+---
+c: Copyright (C) Daniel Stenberg, <daniel.se>, et al.
+SPDX-License-Identifier: curl
+Title: CURLOPT_ECH
+Section: 3
+Source: libcurl
+See-also:
+  - CURLOPT_DOH_URL (3)
+Protocol:
+  - HTTPS
+---
+
+# NAME
+
+CURLOPT_ECH - configuration for Encrypted Client Hello
+
+# SYNOPSIS
+
+~~~c
+#include <curl/curl.h>
+
+CURLcode curl_easy_setopt(CURL *handle, CURLOPT_ECH, char *config);
+~~~
+
+# DESCRIPTION
+
+Pass a string that specifies configuration details for ECH.
+In all cases, if ECH is attempted, it may fail for various reasons.
+The keywords supported are:
+
+## false
+Turns off ECH.
+## grease
+Instructs client to emit a GREASE ECH extension.
+(The connection fails if ECH is attempted but fails.)
+## true
+Instructs client to attempt ECH, if possible, but to not fail if attempting ECH is not possible.
+## hard
+Instructs client to attempt ECH and fail if if attempting ECH is not possible.
+## ecl:\<base64-value\>
+If the string starts with "ecl:" then the remainder of the string should be a base64-encoded
+ECHConfigList that is used for ECH rather than attempting to download such a value from
+the DNS.
+## pn:\<name\>
+If the string starts with "pn:" then the remainder of the string should be a DNS/hostname
+that is used to over-ride the public_name field of the ECHConfigList that is used
+for ECH.
+
+# DEFAULT
+
+NULL, meaning ECH is disabled.
+
+# EXAMPLE
+
+~~~c
+CURL *curl = curl_easy_init();
+
+const char *config ="ecl:AED+DQA87wAgACB/RuzUCsW3uBbSFI7mzD63TUXpI8sGDTnFTbFCDpa+CAAEAAEAAQANY292ZXIuZGVmby5pZQAA";
+if(curl) {
+  curl_easy_setopt(curl, CURLOPT_ECH, config);
+  curl_easy_perform(curl);
+}
+~~~
+
+
+# AVAILABILITY
+
+Added in 8.6.0
```

Comment:
The feature window closes on April 27 for this cycle. So we have two long weeks! :grinning:

Reply:
Well, that timing's not too bad - after the 29th I'll be marking a lot of exam scripts so between now and then is a good time to try get it over the line:-)

I plan to try address all these comments and re-do all the tests (with boring/wolf etc) next few days so whether this makes it will likely depend on whether your or others' reviews find problems mostly.